sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+The resource path handled automatically because of this, and the `SharedKeyCredential` authenticating the client is also used automatically to sign the SAS. See [building a client](#buildclient) for how to get an authenticated client.
+```java
+// specify token properties
+String identifier = "identifier";
+BlobSASPermission permissions = new BlobSASPermission()
+        .read(true)
+        .create(true)
+        .delete(true)
+        .write(true);
+OffsetDateTime startTime = OffsetDateTime.now().minusDays(1);
+OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+IPRange ipRange = new IPRange()
+        .ipMin("0.0.0.0")
+        .ipMax("255.255.255.255");
+SASProtocol sasProtocol = SASProtocol.HTTPS_HTTP;
+String cacheControl = "cache";
+String contentDisposition = "disposition";
+String contentEncoding = "encoding";
+String contentLanguage = "language";
+String contentType = "type";
+String version = Constants.HeaderConstants.TARGET_STORAGE_VERSION;
+
+// build the token
+String sasToken = blobClient.generateSAS(identifier, permissions, expiryTime, startTime, version, sasProtocol,
+            ipRange, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType);
+```
+
+## Build Client
+In v10/v11, core classes were built with constructors, taking a `URL` and an already constructed `HttpPipeline`.
+```java
+final BlockBlobURL blobURL = new BlockBlobURL(
+    new URL("https://" + accountName + ".blob.core.windows.net/mycontainer/myimage.jpg"), 
+    StorageURL.createPipeline(myCredential, new PipelineOptions()));
+```
+
+In v12, we have moved to a builder pattern. To replicate the above v10/v11 snippit:
+```java
+final BlockBlobClient blobClient = new BlobClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net")
+    .containerName("mycontainer")
+    .blobName("myimage.jpg")
+    .credential(myCredential)
+    .buildBlockBlobClient(); // we can also choose PageBlobClient, AppendBlobClient, and BlobClient
+```
+These builders can also build sync and async clients interchangably. Just select the correct "build" method. Additionally, if you already have the full URL, you can just specify that as the endpoint instead of naming container and blob individually, similar to v10/v11.
+```java
+ final BlockBlobClient blobClient = new BlobClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net/mycontainer/myimage.jpg")
+    .credential(myCredential)
+    .buildBlockBlobAsyncClient(); // we choose the "async" method this time
+``` 
+
+## Credentials
+
+### Connection string 
+v10/v11 did not support connection strings. In v12, our builders accept connection strings, no calls to `endpoint()` or `credential()` required. However, connection strings generally only point at the account, so you WILL need to specify container and blob names as necessary.
+```java
+ContainerClient client = new ContainerClientBuilder()
+    .connectionstring("${connectionstring}")
```

Comment:
This syntax might make a user think that this is a valid string, you might want to be more explicit, e.g. `"<CONNECTION_STRING_HERE>"`